app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormatExpr.scala

```diff
@@ -281,6 +281,22 @@ case object EORI extends TextConstraint
 case object UkEORI extends TextConstraint
 case object ChildBenefitNumber extends TextConstraint
 
+final case class PositiveNumberOrNumber(
+  maxWholeDigits: Int = TextConstraint.defaultWholeDigits,
+  maxFractionalDigits: Int = TextConstraint.defaultFractionalDigits,
+  roundingMode: RoundingMode = RoundingMode.defaultRoundingMode,
+  unit: Option[LocalisedString] = None)
+
+final object IsPositiveNumberOrNumber {
+  def unapply(tc: TextConstraint): Option[PositiveNumberOrNumber] = tc match {
```

Comment:
I've thought I've mentioned it before, but I can't see it so here it is:

Can we drop `PositiveNumberOrNumber` case class (it serves no real purpose) and return `Tuple` from this `unapply` instead? This was pattern matching with `IsPositiveNumberOrNumber` will be more readable.